internal/receiver/smartagentreceiver/config.go

```diff
@@ -99,13 +99,7 @@ func (cfg *Config) Unmarshal(componentParser *config.Parser) error {
 	// which is a problem when unmarshalling custom agent monitor configs.  Here we use a map of lowercase to supported
 	// case tag key names and update the keys where applicable.
 	yamlTags := yamlTagsFromStruct(monitorConfigType)
-	for key, val := range allSettings {
-		updatedKey := yamlTags[key]
-		if updatedKey != "" {
-			delete(allSettings, key)
-			allSettings[updatedKey] = val
-		}
-	}
+	recursivelyCapitalizeConfigKeys(allSettings, yamlTags)
```

Comment:
I believe this should fix the issue reported in #322, not being able to set `authType` in `kubelet.APIConfig`.